File: app/src/main/java/jgappsandgames/smartreminderslite/status/StatusActivity.java
package jgappsandgames.smartreminderslite.status;

// Java
import java.util.ArrayList;
import java.util.Calendar;

// Android OS
import android.app.Activity;
import android.os.Bundle;

// Android View
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

// App
import jgappsandgames.smartreminderslite.R;
import jgappsandgames.smartreminderslite.holder.TaskFolderHolder.OnTaskChangedListener;

// Save
import jgappsandgames.smartreminderssave.tasks.Task;
import jgappsandgames.smartreminderssave.tasks.TaskManager;

/**
 * Status Activity
 * Created by joshua on 9/4/17.
 */
public class StatusActivity extends StatusActivityInterface {
    // Data
    private ArrayList<Task> overdue_array;
    private ArrayList<Task> incomplete_array;
    private ArrayList<Task> done_array;

    // LifeCycle Methods
    @Override
    protected void onResume() {
        super.onResume();

        // Load Data
        overdue_array = new ArrayList<>();
        incomplete_array = new ArrayList<>();
        done_array = new ArrayList<>();

        for (int i = 0; i < TaskManager.tasks.size(); i++) {
            Task temp = new Task(TaskManager.tasks.get(i));
            if (temp.getType() == Task.TYPE_TASK) {
                if (temp.getStatus() == Task.STATUS_DONE) done_array.add(temp);
                else if (temp.getDateDue() == null) incomplete_array.add(temp);
                else if (temp.getDateDue().before(Calendar.getInstance())) overdue_array.add(temp);
                else incomplete_array.add(temp);
            }
        }

        // Set Adapters
        overdue_adapter = new StatusAdapter(this, overdue_array);
        incomplete_adapter = new StatusAdapter(this, incomplete_array);
        done_adapter = new StatusAdapter(this, done_array);

        try {
            overdue_list.setAdapter(overdue_adapter);
        } catch (NullPointerException n) {
            n.printStackTrace();
        }

        try {
            incomplete_list.setAdapter(incomplete_adapter);
        } catch (NullPointerException n) {
            n.printStackTrace();
        }

        try {
            done_list.setAdapter(done_adapter);
        } catch (NullPointerException n) {
            n.printStackTrace();
        }
    }

    @Override
    public void onTaskChanged() {
        onResume();
    }

    @Override
    public void save() {
        TaskManager.save();
    }
}

File: app/src/main/java/jgappsandgames/smartreminderslite/sort/status/StatusActivity.kt
package jgappsandgames.smartreminderslite.sort.status

// Android OS
import android.app.Activity
import android.content.Intent
import android.os.Bundle

// Views
import android.view.Menu
import android.view.MenuItem

// App
import jgappsandgames.smartreminderslite.R
import jgappsandgames.smartreminderslite.adapter.TaskAdapterInterface
import jgappsandgames.smartreminderslite.holder.TaskFolderHolder
import jgappsandgames.smartreminderslite.home.FirstRun
import jgappsandgames.smartreminderslite.utility.OptionsUtility

// KotlinX
import kotlinx.android.synthetic.main.activity_status.status_done_list
import kotlinx.android.synthetic.main.activity_status.status_done_text
import kotlinx.android.synthetic.main.activity_status.status_incomplete_list
import kotlinx.android.synthetic.main.activity_status.status_incomplete_text
import kotlinx.android.synthetic.main.activity_status.status_overdue_list
import kotlinx.android.synthetic.main.activity_status.status_overdue_text

// Save
import jgappsandgames.smartreminderssave.MasterManager
import jgappsandgames.smartreminderssave.status.StatusManager
import jgappsandgames.smartreminderssave.tasks.Task
import jgappsandgames.smartreminderssave.utility.FileUtility

/**
 * StatusActivity
 * Created by joshua on 12/14/2017.
 */
class StatusActivity: Activity(), TaskFolderHolder.OnTaskChangedListener {
    // LifeCycle Methods ---------------------------------------------------------------------------
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_status)

        // Handle Data
        FileUtility.loadFilePaths(this)
        if (FileUtility.isFirstRun()) startActivity(Intent(this, FirstRun::class.java))
        else MasterManager.load()
    }

    override fun onResume() {
        super.onResume()

        // Create Status Manager
        StatusManager.create()

        val i = ArrayList<Task>()
        i.addAll(StatusManager.getInProgress())
        i.addAll(StatusManager.getIncomplete())

        // Set Adapters
        status_overdue_list.adapter = StatusAdapter(this, StatusManager.getOverdue())
        status_incomplete_list.adapter = StatusAdapter(this, i)
        status_done_list.adapter = StatusAdapter(this, StatusManager.getCompleted())

        // Set Text
        status_overdue_text.text = getString(R.string.overdue_tasks, StatusManager.getOverdue().size)
        status_incomplete_text.text = getString(R.string.incomplete_tasks, i.size)
        status_done_text.text = getString(R.string.completed_tasks, StatusManager.getCompleted().size)
    }

    // Menu Methods --------------------------------------------------------------------------------
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_auxilary, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return OptionsUtility.onOptionsItemSelected(this, item, object: OptionsUtility.Save {
            override fun save() {
                this@StatusActivity.save()
            }
        })
    }

    // Task Change Listeners -----------------------------------------------------------------------
    override fun onTaskChanged() {
        onResume()
    }

    // Parent Methods ------------------------------------------------------------------------------
    fun save() {
        MasterManager.save()
    }

    // Internal Classes ----------------------------------------------------------------------------
    class StatusAdapter(activity: StatusActivity, tasks: ArrayList<Task>):
            TaskAdapterInterface(activity, activity, tasks)
}